source/slang/slang-ast-base.h

```diff
@@ -23,39 +23,49 @@ class NodeBase : public RefObject
 {
     SLANG_ABSTRACT_CLASS(NodeBase)
 
-        // By default AST types do *not* store the builder. This is called when constructed tho.
-    SLANG_FORCE_INLINE void setASTBuilder(ASTBuilder* astBuilder) { SLANG_UNUSED(astBuilder); }
+        // MUST be called before used. Called automatically via the ASTBuilder.
+        // Note that the astBuilder is not stored in the NodeBase derived types by default.
+    SLANG_FORCE_INLINE void init(ASTNodeType inAstNodeType, ASTBuilder* /* astBuilder*/ ) { astNodeType = inAstNodeType; }
```

Comment:
Unrelated to this change, but: I'd really like us to decide on and document a policy for how to name "tag" types like `ASTNodeType`.

I'm not a fan of using "type" or "kind" for these, and have usually used "flavor" for the ones I introduce (because the terms "type" and "kind" have more specialized meaning in a compiler or type system). That said, I also know that using "type" for this kind of field is something like the 99% case for codebases in the wild.

Using `astNodeType` instead of just `type` is a nice step because it makes it clear that it represents the type of the AST node object and not something like the type of an expression (which is represented as an AST node...).

There's nothing to change in this PR, so this is just my thoughts meandering...